pkg/kubelet/kubelet.go

```diff
@@ -1469,8 +1470,33 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	podStatus := o.podStatus
 	updateType := o.updateType
 
+	//Get the hostname for traces
+	activeHost := string(kl.nodeName)
+
+	// Create and register an OpenCensus
+	// Stackdriver Trace exporter.
+	exporter, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID: "samnaser-gke-dev-217421",
+	})
+	if err != nil {
+		return fmt.Errorf("error configuring stackdriver opencensus exporter: %v", err)
+	}
+
+	// Register our exporter with our tracer and set a liberal sampling policy for testing
+	trace.RegisterExporter(exporter)
+	trace.ApplyConfig(trace.Config{DefaultSampler: trace.ProbabilitySampler(0.15)})
+
+	// Parent span, annotate it with the hostname
+	ctx := context.Background()
+	ctx, span := trace.StartSpan(ctx, "Kubelet: SyncPod")
+	span.AddAttributes(trace.StringAttribute("host", activeHost))
+	defer span.End()
+
 	// if we want to kill a pod, do it now!
 	if updateType == kubetypes.SyncPodKill {
+
+		_, podKillSpan := trace.StartSpan(ctx, "Pod kill logic")
```

Comment:
Don't use regular sentences as span names.

Span names are statistically interesting identifier to represent a task.

For example,

```
_, podKillSpan := trace.StartSpan(ctx, "kubernetes.podkill")
```
would be a better fit. The comment applies to all the spans.

Reply:
This is the right way to construct a span name.